pkg/gui/discard_changes_menu_panel.go

```diff
@@ -40,10 +40,16 @@ func (gui *Gui) handleCreateDiscardMenu(g *gocui.Gui, v *gocui.View) error {
 			{
 				displayString: gui.Tr.LcDiscardAllChanges,
 				onPress: func() error {
-					if err := gui.GitCommand.DiscardAllFileChanges(file); err != nil {
-						return gui.surfaceError(err)
-					}
-					return gui.refreshSidePanels(refreshOptions{mode: ASYNC, scope: []int{FILES}})
+					return gui.ask(askOpts{
```

Comment:
I would add a config key to allow the user to configure whether they want to confirm here. I personally would find it tedious to have to press three keys every time I wanted to delete a file. What are your thoughts on adding a function like the following to only conditionally ask for confirmation:
```go
func (gui *Gui) conditionalAsk(skip bool, opts askOpts) error {
	if skip {
		if opts.handleConfirm != nil {
			return gui.surfaceError(opts.handleConfirm())
		}
		return nil
	}

	return gui.ask(opts)
}
```

Then we can introduce a new config key called SkipDiscardChangesWarning in the GuiConfig in `pkg/config/user_config.go`:
```
SkipDiscardChangesWarning bool               `yaml:"skipDisacrdChangesWarning"`
```

And then we can combine the two together like so:
```
return gui.conditionalAsk(
						gui.Config.GetUserConfig().Gui.SkipDiscardChangesWarning,
						askOpts{
							title:  gui.Tr.ConfirmationTitle,
							prompt: gui.Tr.DiscardAllChangesPrompt,
							handleConfirm: func() error {
								if err := gui.GitCommand.DiscardAllFileChanges(file); err != nil {
									return gui.surfaceError(err)
								}
								return gui.refreshSidePanels(refreshOptions{mode: ASYNC, scope: []int{FILES}})
							},
						})
```

We would need to decide on a sensible default here: should we default to showing these warnings or not? If we do show the warnings, more people will feel safe using lazygit, but it will also slow people down. I'm leaning towards having the warning there by default.

Lemme know your thoughts :)